source/features/feature-descriptions.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'dom-chef';
+import domify from 'doma';
+import elementReady from 'element-ready';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {wrapAll} from '../helpers/dom-utils';
+import GitHubURL from '../github-helpers/github-url';
+import parseBackticks from '../github-helpers/parse-backticks';
+
+interface Commit {
+	messageHeadline: string;
+	messageHeadlineHTML: string;
+	commitUrl: string;
+	committedDate: string;
+}
+
+const getFeatureHistory = async (fileName: string): Promise<Commit[]> => {
+	const {repository} = await api.v4(`
+		repository() {
+			defaultBranchRef {
+				target {
+					...on Commit {
+						history(first:100, path: "source/features/${fileName}") {
+							nodes {
+								messageHeadline
+								messageHeadlineHTML
+								commitUrl
+								committedDate
+							}
+						}
+					}
+				}
+			}
+		}
+	`);
+	return repository.defaultBranchRef.target.history.nodes;
+};
+
+function getCommitUrl(commit: Commit): string {
+	const [, pullRequestUrl] = /<a[^>]+href="([^"]+)">/.exec(commit.messageHeadlineHTML) ?? [];
+	return pullRequestUrl ?? commit.commitUrl;
+}
+
+async function getHistoryDropdown(featureName: string): Promise<Element> {
+	let history = await getFeatureHistory(featureName + '.tsx');
```

Comment:
I think we should add a detection to featureMeta if it's a CSS feature. Avoiding 2 calls.

Reply:
Also it needs a cache.function...

Reply:
Problem is: when visiting a `.css` file, there is no way to tell (without an API call) whether it belongs to a regular `.tsx` feature, or a CSS-only `.ts` feature.

Reply:
> I think we should add a detection to featureMeta

Add it here

Reply:
> add a detection to featureMeta

How? `__featuresMeta__` is provided by `webpack.config.ts`, which only loads `source/refined-github.ts` and `readme.md`.

Reply:
Can we detect if the file has `features.addCssFeature`?

Reply:
That only works on CSS features that can be disabled. If you’re visiting a real CSS-only feature it won’t appear in __featuresMeta__ at all. 

Does `tsx` have to be specified in the API call?

Reply:
That only works on CSS features that can be disabled. If you’re visiting a real CSS-only feature it won’t appear in __featuresMeta__ at all. 

Does `tsx` have to be specified in the API call?

Reply:
Duhh

Reply:
> Does `.tsx` have to be specified in the API call? Can it specify both tsx and css?

It expects exactly one full file path including file extension, or none at all.

> Maybe this information box just can’t appear on CSS features.

I fear you might be right :frowning_face:

Reply:
I feel that this feature is replacing the description and picture that we removed. The CSS ones never had it until now.